specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2024-04-01/mfe.json

```diff
@@ -12140,6 +12180,47 @@
       },
       "additionalProperties": false
     },
+    "BatchPipelineComponentDeploymentConfiguration": {
+      "description": "Properties for a Batch Pipeline Component Deployment.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BatchDeploymentConfiguration"
+        }
+      ],
+      "properties": {
+        "componentId": {
+          "description": "The ARM id of the component to be run.",
+          "$ref": "#/definitions/IdAssetReference",
+          "x-nullable": true
+        },
+        "description": {
+          "description": "The description which will be applied to the job.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "settings": {
```

Comment:
Dictionaries are not allowed in Azure unless they satisfy the following criteria. Otherwise, it should be defined as a model or model array.

1. The schema is not owned by your service
2. The keys of the dictionary are not validated by your service
3. The property represents an opaque property that is not interpreted by your service and is purely a pass thru.

Reply:
I just bring the changes of this PR: https://github.com/Azure/azure-rest-api-specs/commit/913cc602374577d6cf2f2fdef4ce3c30a9c46bd4 back.